src/test/java/org/apache/commons/compress/utils/IOUtilsTest.java

```diff
@@ -148,6 +148,25 @@ public void readRangeFromChannelDoesntReadMoreThanAskedFor() throws IOException
         }
     }
 
+    @Test
+    public void readRangeFromChannelDoesntReadMoreThanAskedForWhenItGotLessInFirstReadCall() throws IOException {
+        try (ReadableByteChannel in = new SeekableInMemoryByteChannel(new byte[] { 1, 2, 3, 4, 5 }) {
+            @Override
+            public int read(ByteBuffer buf) throws IOException {
+                // Read max 3 bytes at a time
+                ByteBuffer temp = ByteBuffer.allocate(Math.min(3, buf.remaining()));
```

Comment:
I think we can have a `final` here.
Even through this is just a test, it's always appreciated if `final` is always used if possible.